monai/bundle/config_parser.py

```diff
@@ -0,0 +1,224 @@
+        net = parser.get_parsed_content("my_net", instantiate=True)
+        print(net)
+
+        # also support to get the configuration content of parsed `ConfigItem`
+        trainer = parser.get_parsed_content("trainer", instantiate=False)
+        print(trainer)
+
+    Args:
+        config: input config source to parse.
+        excludes: when importing modules to instantiate components,
+            excluding components from modules specified in ``excludes``.
+        globals: pre-import packages as global variables to ``ConfigExpression``,
+            so that expressions, for example, ``"$monai.data.list_data_collate"`` can use ``monai`` modules.
+            The current supported globals and alias names are
+            ``{"monai": "monai", "torch": "torch", "np": "numpy", "numpy": "numpy"}``.
+            These are MONAI's minimal dependencies.
+
+    See also:
+
+        - :py:class:`monai.apps.ConfigItem`
+
+    """
+
+    def __init__(
+        self,
+        config: Any,
+        excludes: Optional[Union[Sequence[str], str]] = None,
+        globals: Optional[Dict[str, Any]] = None,
+    ):
+        self.config = None
+        self.globals: Dict[str, Any] = {}
+        globals = {"monai": "monai", "torch": "torch", "np": "numpy", "numpy": "numpy"} if globals is None else globals
+        if globals is not None:
+            for k, v in globals.items():
+                self.globals[k] = importlib.import_module(v) if isinstance(v, str) else v
+
+        self.locator = ComponentLocator(excludes=excludes)
+        self.ref_resolver = ReferenceResolver()
+        self.set(config=config)
+
+    def __repr__(self):
+        return f"{self.config}"
+
+    def __getitem__(self, id: Union[str, int]):
+        """
+        Get the config by id.
+
+        Args:
+            id: id of the ``ConfigItem``, ``"#"`` in id are interpreted as special characters to
+                go one level further into the nested structures.
+                Use digits indexing from "0" for list or other strings for dict.
+                For example: ``"xform#5"``, ``"net#<args>#channels"``. ``""`` indicates the entire ``self.config``.
+
+        """
+        if id == "":
+            return self.config
+        config = self.config
+        for k in str(id).split(self.ref_resolver.sep):
+            if not isinstance(config, (dict, list)):
+                raise ValueError(f"config must be dict or list for key `{k}`, but got {type(config)}: {config}.")
+            indexing = k if isinstance(config, dict) else int(k)
+            config = config[indexing]
+        return config
+
+    def __setitem__(self, id: Union[str, int], config: Any):
+        """
+        Set config by ``id``.  Note that this method should be used before ``parse()`` or ``get_parsed_content()``
+        to ensure the updates are included in the parsed content.
+
+        Args:
+            id: id of the ``ConfigItem``, ``"#"`` in id are interpreted as special characters to
```

Comment:
Hi @wyli ,

I saw you made the `sep` configurable in `ReferenceResolver`, do you want to change all the doc-string for the "#" description?

Thanks.

Reply:
I think for now `#` is more readable as part of the docstring